src/DurableTask.Netherite/StorageLayer/Faster/FasterKV.cs

```diff
@@ -968,8 +968,8 @@ async Task ReadIssueLoop()
                 {
                     while (enumerator.MoveNext())
                     {
-                        if (!string.IsNullOrEmpty(instanceQuery?.InstanceIdPrefix)
-                            && !enumerator.Current.StartsWith(instanceQuery.InstanceIdPrefix))
+                        if ((!string.IsNullOrEmpty(instanceQuery?.InstanceIdPrefix) && !enumerator.Current.StartsWith(instanceQuery.InstanceIdPrefix))
+                            || (instanceQuery.ExcludeEntities && DurableTask.Core.Common.Entities.IsEntityInstance(enumerator.Current)))
```

Comment:
should this be:

```suggestion
                            || (!instanceQuery.ExcludeEntities && DurableTask.Core.Common.Entities.IsEntityInstance(enumerator.Current)))
```

?

Reply:
I think it is correct as it is now. The conditional determines whether the record should be excluded from the results. So the condition has to return true iff  `ExcludeEntities` is true and the instance is an entity.